### ISSUE TITLE
RedundantModifierCheck: redundant public modifier on top-level method not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20266]
Parent Issue #19971
Split off from #20226 (that issue is scoped to the `final` modifier; this one tracks `public`).

I have read check documentation: https://checkstyle.org/checks/modifier/redundantmodifier.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="RedundantModifier"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputRedundantModifierPublicCompact.java <<'EOF'
public void myMethod() { }
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputRedundantModifierPublicCompact.java

Starting audit...
Audit done.
```
---

**Describe what you expect in detail.**

RedundantModifierCheck must flag `public` on a top-level method in a JEP 512 compact source file. The implicit class in a compact source file is final and it lives in the unnamed package, so it can never be referenced by another class. That means public on its methods can't widen access to anything, it's just obsolete.

Expected on the compact input: 1 violation `Redundant 'public' modifier.` on `myMethod`.

**Comment by stoyanK7** [comment]
@mahfouz72 Could you please review this issue and put the `approved` label if you agree? I personally agree.

**Comment by mahfouz72** [comment]
> RedundantModifierCheck must flag public on a top-level method in a JEP 512 compact source file. The implicit class in a compact source file is final and it lives in the unnamed package, so it can never be referenced by another class. That means public on its methods can't widen access to anything, it's just obsolete.

I would lean toward yes, it is reasonable to report public as redundant.  There is effectively no source-level way for another compilation unit to access myMethod(). The enclosing class is implicitly declared and not intended to be referenced by name

**Comment by romani** [comment]
The same is for all visibility modifiers , right?
